### PR TITLE
Add RFC for role change process

### PR DESCRIPTION
**Pull request by Gerg**
I'm not particularly attached to the criteria I specified. Feel free to suggest major modifications. I'm just trying to spark conversation around this process.

### CODE REVIEW

File: toc/rfc/rfc-draft-role-change-process.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+### Revoking Contributor Role
+
+- People with the Contributor role may submit a PR to revoke their role by
+  removing the appropriate entry from
+  [cloudfoundry.yml](https://github.com/cloudfoundry/community/blob/main/org/cloudfoundry.yml).
+
+- An existing Contributor or Approver may submit the revocation request on
+  behalf of someone else, but the person whose role is being revoked must be
+  given a week to refute the revocation.
```
Comment (tcdowney) [inline toc/rfc/rfc-draft-role-change-process.md:77]: Maybe these should be bumped to two weeks to better handle someone being away on vacation, leave, etc.

File: toc/rfc/rfc-draft-role-change-process.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  by reviewing the PR.
+
+- For Working Groups with fewer than 4 approvers, a single Approver review is
+  sufficient.
+
+- An existing Approver may submit the promotion request on behalf of someone else, but they
+  may not serve as a reviewer.
+
+- A Working Group Lead for that Working Group may merge the PR after it has been
+  reviewed.
```
Comment (ameowlia) [inline toc/rfc/rfc-draft-role-change-process.md:64]: Can you clarify what you mean by "may" here? I'm not sure which of these two options are you implying...

1.  A WG lead _may_ merge it or _may_ choose not to merge it for some reason?
OR
2. That a WG lead _may_ merge the PR or that someone else _may_ do it?

I am onboard with option 1 and not option 2.

File: toc/rfc/rfc-draft-role-change-process.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  removing the appropriate entry from
+  [cloudfoundry.yml](https://github.com/cloudfoundry/community/blob/main/org/cloudfoundry.yml).
+
+- An existing Contributor or Approver may submit the revocation request on
+  behalf of someone else, but the person whose role is being revoked must be
+  given a week to refute the revocation.
+
+- Any Working Group Lead may merge the PR, without review.
+
+- TOC members may merge the PR at their discretion.
```
Comment (ameowlia) [inline toc/rfc/rfc-draft-role-change-process.md:81]: What is the difference between how WG leads and toc members should act here? These sound like the same thing written two different ways? Or is there a distinction?

File: toc/rfc/rfc-draft-role-change-process.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+### Revoking Approver Role
+
+- People with an Approver role may submit a PR to revoke their role by removing
+  the appropriate entry from
+  [cloudfoundry.yml](https://github.com/cloudfoundry/community/blob/main/org/cloudfoundry.yml).
```
Comment (ameowlia) [inline toc/rfc/rfc-draft-role-change-process.md:87]: Same as above

```suggestion
- People with an Approver role may submit a PR to revoke their role by removing
  the appropriate entry from the yaml definition in their Working Group's charter.
```

File: toc/rfc/rfc-draft-role-change-process.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+difficult for community members to know how to apply for a role promotion and
+difficult for Working Group leadership to manage Contributor and Approver
+candidates.
+
+In addition, there is no defined process for removing people from Contributor or
+Approver roles if they are no longer interested in participating at that level.
+
+## Proposal
+
+### Promotion to Contributor
```
Comment (stephanme) [inline toc/rfc/rfc-draft-role-change-process.md:29]: Overlaps to a good extend https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0002-github-members.md

Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md:29]: I view this as an extension of that RFC. I'm primarily interested in the process through which the promotion is reviewed and accepted.

That said, maybe there is more I can do to integrate the two RFCs.

Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md:29]: For instance, I think this RFC currently conflicts with this paragraph:


> As removing a member from the `cloudfoundry` GitHub org may have unintended consequences across the organization, the TOC is the body required to approve those removals.
Proposals to remove a member should also be submitted via PR, and the submitter should tag the PR with `toc` and should mention `@cloudfoundry/toc` to make the TOC aware of the request.
The TOC will then consult any working groups that may be affected by the removal of this member and use its usual decision process to approve or reject the removal.

File: toc/rfc/rfc-draft-role-change-process.md (hunk 1 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md]: Addressed by https://github.com/cloudfoundry/community/pull/248/commits/86ce2597a1d6e299461f640e6ba78a9c63f683dc

File: toc/rfc/rfc-draft-role-change-process.md (hunk 3 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md:81]: I think the distinction I'm gesturing towards that I haven't figured out how to codify is something like:
- Working Group Leads are obligated to merge the role change requests (or reject the request, as appropriate)
- TOC members can use their ultimate authority to intervene in special circumstances, however this is not the expected workflow

Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md:81]: For this particular case, it looks like the TOC might need to be involved based on https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0002-github-members.md.

Not sure if that is desirable/scalable. Especially if we want to start doing a better job removing lapsed contributors.

File: toc/rfc/rfc-draft-role-change-process.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+- TOC members may bypass the review process and merge the PR at their
+  discretion.
+
+### Promotion to Approver
+
+- When a person meets the criteria to be an Approver for a Working Group as defined in
+[ROLES.md](https://github.com/cloudfoundry/community/blob/main/toc/ROLES.md),
+they may submit a PR adding themselves to the appropriate Working Group in
+[cloudfoundry.yml](https://github.com/cloudfoundry/community/blob/main/org/cloudfoundry.yml).
```
[the remark this answers is not in the document]
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md]: Addressed by https://github.com/cloudfoundry/community/pull/248/commits/994b3a52a9b76e79083a3bf91969dae29a47442a

File: toc/rfc/rfc-draft-role-change-process.md (hunk 4 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md]: Addressed by https://github.com/cloudfoundry/community/pull/248/commits/994b3a52a9b76e79083a3bf91969dae29a47442a

File: toc/rfc/rfc-draft-role-change-process.md (hunk 2 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md:64]: Maybe something like:
> A Working Group Lead for that Working Group _will_ merge or close the PR based on the results of the review and their discretion.

File: toc/rfc/rfc-draft-role-change-process.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+Approver roles if they are no longer interested in participating at that level.
+
+## Proposal
+
+### Promotion to Contributor
+
+- When a person meets the criteria to be a Contributor as defined in
+[ROLES.md](https://github.com/cloudfoundry/community/blob/main/toc/ROLES.md),
+they may submit a PR adding themselves to the organization in
+[cloudfoundry.yml](https://github.com/cloudfoundry/community/blob/main/org/cloudfoundry.yml).
```
Comment (rkoster) [inline toc/rfc/rfc-draft-role-change-process.md:34]: the working group charter should be the source of truth when it comes to roles (Approver and above).

File: toc/rfc/rfc-draft-role-change-process.md (hunk 5 above)
Comment (stephanme) [reply toc/rfc/rfc-draft-role-change-process.md:29]: I don't remember the exact reason why removing a member from the cloudfoundry org might have unintended consequences. Can't really imaging for a contributor and I agree that TOC should not be required here.

Maybe this discussion is caused by using a low level technical document (cloudfoundry.yml) for maintaining a higher level concept (contributor role).  Could be solved by having a separate contributor document that gets (eventually automated) translated into cloudfoundry.yml similar to the WG charters. Allows e.g. to ensure that all approvers are members of the cloudfoundry org and makes it easier for new contributors to file a PR: cloudfoundry.yml has currently 7k lines and ~650 org members.

**Comment by dsboulder** [comment]
Is this ready for the approval period at this point?

File: toc/rfc/rfc-draft-role-change-process.md (hunk 7 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md:35]: Addressed in https://github.com/cloudfoundry/community/pull/248/commits/16783a0d52e15c523b7f05e0d8eab200e08705a1

File: toc/rfc/rfc-draft-role-change-process.md (hunk 3 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md]: Addressed in https://github.com/cloudfoundry/community/pull/248/commits/38cd8bd0073be3f8bdaf2757451f098235afbc8e

File: toc/rfc/rfc-draft-role-change-process.md (hunk 2 above)
Comment (Gerg) [reply toc/rfc/rfc-draft-role-change-process.md]: Addressed in https://github.com/cloudfoundry/community/pull/248/commits/38cd8bd0073be3f8bdaf2757451f098235afbc8e

**Comment by Gerg** [comment]
@dsboulder I've made all the changes I intend to make, so should be ready to go.

**Comment by emalm** [comment]
Sounds like @Gerg is moving to start the Final Comment Period on this proposed RFC. @cloudfoundry/toc, please vote (👍 / 👎 ) on this comment, or otherwise comment on this conversation!

**Comment by rkoster** [comment]
org/cloudfoundry.yml should be an implementation detail. I would also vote for a separate contributor.yml file.

**Comment by stephanme** [comment]
As discussed in the last TOC meeting: I amended the PR and introduce a dedicated contributors.yml (initialized with the org members copied from cloudfoundry.yml).

An automation could look like this:
- cloudfoundry.yml keeps the static content that is not generated (org config, projects but not anymore org members and eventually no teams)
- org members are generated from contributors.yml + all users mentioned in the WG charters and TOC.md
- teams can be generated from WG charters and TOC.md
- cloudfoundry.yml, org members and teams are merged into one file that is the input for Peribolos

**Comment by stephanme** [comment]
After the changes I guess we should start a new Final Comment Period on this proposed RFC. https://github.com/orgs/cloudfoundry/teams/toc, please vote (👍 / 👎 ) on this comment, or otherwise comment on this conversation!

**Review by stephanme — APPROVED** [review]

**Review by emalm — APPROVED** [review]

**Review by LeePorte — APPROVED** [review]

**Pull request merged**